bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java

```diff
@@ -549,7 +540,8 @@ private void updateThermostatState(NhcThermostat2 thermostat, NhcDevice2 device)
         Optional<NhcProperty> ambientTemperatureProperty = device.properties.stream()
                 .filter(p -> (p.ambientTemperature != null)).findFirst();
 
-        Optional<NhcProperty> demandProperty = device.properties.stream().filter(p -> (p.demand != null)).findFirst();
+        Optional<NhcProperty> demandProperty = device.properties.stream()
+                .filter(p -> ((p.demand != null) || (p.operationMode != null))).findFirst();
```

Comment:
Are you sure about this change? This can result in a property where `demand` can be null and in l. 572 you switch on `demand` from that property without a default

Reply:
Thanks for the catch. Yes, I am sure about this, but I need to also extend the code around line 572 for it.